log/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Warning;
+}
+
+/// Marker for URID representing the nature of a log message.
+// Note : it's may be better to have a URID trait to define a common interface
+pub unsafe trait EntryType
+where
+    Self: Copy,
+{
+}
+
+unsafe impl EntryType for URID<ErrorClass> {}
+unsafe impl EntryType for URID<NoteClass> {}
+unsafe impl EntryType for URID<TraceClass> {}
+unsafe impl EntryType for URID<WarningClass> {}
+
+/// Errors potentially generated by [`Log`](struct.Log.html) methods.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum LogError {
+    /// An error occured when sending message to the host
+    PrintError,
+    /// The provided string don't have a `'\0'` to indicate it's end
+    NoNullTerminator,
+    /// No callback was provided by the host
+    ///
+    /// This can only happen with a faulty host
+    NoCallback,
+}
+
+/// The Log feature.
+#[repr(transparent)]
+pub struct Log<'a> {
+    internal: &'a lv2_sys::LV2_Log_Log,
+}
+
+unsafe impl<'a> UriBound for Log<'a> {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__log;
+}
+
+unsafe impl<'a> Feature for Log<'a> {
+    // Note: this feature can be used in any threading class but:
+    // * i have a doubt about it's thread safety, can we assume the host provide this thread safety
+    // since this feature can be used anywhere ?.
+    // * i shouldn't be used in context where rt is a concern, that mean is audiothreadclass in
+    // practice, but it's acceptable to use it for debugging purpose
+    // So, at this time, i just giving access to it instanciation class
+    unsafe fn from_feature_ptr(feature: *const c_void, class: ThreadingClass) -> Option<Self> {
+        match class {
+            ThreadingClass::Audio => {
+                panic!("The log feature is not allowed in the audio threading class")
+            }
+            _ => (feature as *const lv2_sys::LV2_Log_Log)
+                .as_ref()
+                .map(|internal| Self { internal }),
+        }
+    }
+}
+
+impl<'a> Log<'a> {
+    /// Send a log message to the host.
+    ///
+    /// The `entry_type` parameter is an URID representing the kind of log message. There are four
+    /// kind of message:
+    /// * **note:** an informative message.
+    /// * **warning:** a warning message.
+    /// * **error:** an error message.
+    /// * **trace:** a debugging trace. These entries should not be displayed during normal
+    /// operation, but the host may implement an option to display them for debugging purposes.
+    /// This entry type is special in that it may be written to in a real-time thread. It is
+    /// assumed that if debug tracing is enabled, real-time considerations are not a concern.
+    pub fn print(&self, entry_type: URID<impl EntryType>, message: &str) -> Result<(), LogError> {
+        let printf = if let Some(printf) = self.internal.printf {
+            printf
+        } else {
+            return Err(LogError::NoCallback);
```

Comment:
will try to do it later, doing this require change in internal feature representation.

Reply:
@prokopyl this change seems more tricky than i initially thought: 
 
 - I wanted to store only part of feature i really use, but...
 -  i realized i was assuming the feature was constant for the plugin lifetime.
 - I checked the spec about it and i found feature always appeared `const LV2_Feature *const *features`, letting me think feature was effectively constant but...
 -  I remembered a trap about the C/C++ `const` keywords, it's doesn't mean "constant", it just means "you can't change this". In other words, it can be modified elsewhere

Reply:
Finally done, i replaced the `lv2_sys::LV2_Log_Log` by an equivalent without the Option<>